### PR TITLE
[pvr][keymaps] Cleanup: Implement PVR 'direct channel input' and 'channel prev…

### PR DESCRIPTION
**Pull request by ksooo**
…iew' actions/keybindings using virtual windows.

I was finally able to get rid of the PVR dependency of `CGUIWindowFullscreen` and `CGUIWindowVisualisation`.

PVR features 'direct channel number input' and 'channel preview OSD' require special key input handling, which is now implemented in a clean way using virtual windows 'on top' of the real video and music fullscreen windows these features are associated with. No more hardcoded PVR dependencies in the related full screen window classes and abilty to define the actions for the keys used for these features like we always do - using keyboard.xml and friends.

The concept of virtual windows is not new - it is already used for other purposes, for instance numeric seek input while playing videos/music(?). I just discovered and understood it and it came to my mind that this is the right solution for a problem that bugged me for a long time. ;-)

@Jalle19 @FernetMenta mind taking a look.

### CODE REVIEW

**Comment by ksooo** [comment]
jenkins build this please

**Comment by da-anda** [comment]
fun fact: I added the virtual window keymap stuff in the early PVR days to get rid of the (back then) hardcoded button maps for channel up/down. Glad to see it's usefull for more PVR stuff now :)

**Review by Jalle19 — APPROVED** [review]
Don't know this code very well, so

File: xbmc/input/WindowTranslator.cpp [hunk 1]
```diff
@@ -278,18 +281,32 @@ int CWindowTranslator::GetVirtualWindow(int windowId)
     // check if we're in a DVD menu
     if (g_application.GetAppPlayer().IsInMenu())
       return WINDOW_VIDEO_MENU;
-    // check for LiveTV and switch to it's virtual window
-    else if (CServiceBroker::GetPVRManager().IsStarted() && g_application.CurrentFileItem().HasPVRChannelInfoTag())
-      return WINDOW_FULLSCREEN_LIVETV;
+    // special casing for Live TV
+    else if (g_application.CurrentFileItem().HasPVRChannelInfoTag())
+    {
+      if (g_infoManager.IsPlayerChannelInputActive())
```
Comment (FernetMenta) [inline xbmc/input/WindowTranslator.cpp]: imo CWindowTranslator should not depend on GUIInfoManager. GUIInfoManager is a cache and proxy for gui controls.

Comment (ksooo) [reply xbmc/input/WindowTranslator.cpp]: Okay, then we need to keep the pvr dependency in this method. Would that be acceptable?

Comment (FernetMenta) [reply xbmc/input/WindowTranslator.cpp]: Yes, imo better than gui

Comment (ksooo) [reply xbmc/input/WindowTranslator.cpp]: Okay, will change that.

**Comment by ksooo** [comment]
@FernetMenta I incorporated the changes you suggested.

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
jenkins build this please

**Review by FernetMenta — APPROVED** [review]

**Pull request merged**